huaweicloud/services/eps/resource_huaweicloud_enterprise_project.go

```diff
@@ -0,0 +1,205 @@
+	}
+
+	d.SetId(project.ID)
+
+	return resourceEnterpriseProjectRead(ctx, d, meta)
+}
+
+func resourceEnterpriseProjectRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	epsClient, err := config.EnterpriseProjectClient(config.GetRegion(d))
+
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud EPS client : %s", err)
+	}
+
+	project, err := enterpriseprojects.Get(epsClient, d.Id()).Extract()
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "Error retrieving HuaweiCloud Enterprise Project")
+	}
+
+	var enable bool
+	if project.Status == 1 {
+		enable = true
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("name", project.Name),
+		d.Set("description", project.Description),
+		d.Set("type", project.Type),
+		d.Set("status", project.Status),
+		d.Set("enable", enable),
+		d.Set("created_at", project.CreatedAt),
+		d.Set("updated_at", project.UpdatedAt),
+	)
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("error setting HuaweiCloud enterprise project fields: %w", err)
+	}
+
+	return nil
+}
+
+func resourceEnterpriseProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	epsClient, err := config.EnterpriseProjectClient(config.GetRegion(d))
+
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud EPS client : %s", err)
+	}
+
+	if d.HasChange("enable") {
+		var actionOpts enterpriseprojects.ActionOpts
+		if d.Get("enable").(bool) {
+			actionOpts.Action = "enable"
+		} else {
+			actionOpts.Action = "disable"
+		}
+
+		_, err = enterpriseprojects.Action(epsClient, actionOpts, d.Id()).Extract()
+		if err != nil {
+			return fmtp.DiagErrorf("Error enabling/disabling HuaweiCloud Enterprise Project: %s", err)
+		}
+
+	}
+
+	if d.HasChanges("name", "description", "type") {
+		updateOpts := enterpriseprojects.CreateOpts{
+			Name:        d.Get("name").(string),
+			Description: d.Get("description").(string),
+			Type:        d.Get("type").(string),
+		}
+
+		_, err = enterpriseprojects.Update(epsClient, updateOpts, d.Id()).Extract()
+
+		if err != nil {
+			return fmtp.DiagErrorf("Error updating HuaweiCloud Enterprise Project: %s", err)
+		}
+	}
+
+	return resourceEnterpriseProjectRead(ctx, d, meta)
+}
+
+func resourceEnterpriseProjectDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	epsClient, err := config.EnterpriseProjectClient(config.GetRegion(d))
+
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud EPS client : %s", err)
+	}
+
+	actionOpts := enterpriseprojects.ActionOpts{
+		Action: "disable",
+	}
+
+	_, err = enterpriseprojects.Action(epsClient, actionOpts, d.Id()).Extract()
+	if err != nil {
+		return fmtp.DiagErrorf("Error disabling HuaweiCloud Enterprise Project: %s", err)
+	}
+
+	d.SetId("")
```

Comment:
seems we don't need to explicitly set id to ""

Reply:
Yes, it's not necessary.
I'm just following the convention. Neet to remove it?

Reply:
I'm OK with that